Applications/Spire/Source/BookView/BboBox.cpp

```diff
@@ -0,0 +1,135 @@
+#include "Spire/BookView/BboBox.hpp"
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Spire/FieldValueModel.hpp"
+#include "Spire/Spire/ToTextModel.hpp"
+#include "Spire/Ui/Box.hpp"
+#include "Spire/Ui/Layouts.hpp"
+#include "Spire/Ui/TextBox.hpp"
+
+using namespace boost;
+using namespace Nexus;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  const auto BORDER_TOP_COLOR = QColor(0xC8C8C8);
+  const auto DOWNTICK_BORDER_TOP_COLOR = QColor(0xE63F44);
+  const auto UPTICK_BORDER_TOP_COLOR = QColor(0x1FD37A);
+
+  template<typename T, typename U>
+  auto make_bbo_label(std::shared_ptr<QuoteModel> quote, U field) {
+    auto label = make_label(
+      make_to_text_model<T>(make_field_value_model(quote, field),
+        [] (const auto& value) {
+          return to_text(value);
+        }));
+    label->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+    return label;
+  }
+
+  auto get_gap_width(int font_size, int parent_width) {
+    return std::clamp(0.48276 * parent_width - 6.89 * font_size,
```

Comment:
Remove magic numbers here and give each constant a name. As is I have no clue what these numbers are supposed to be.